File: xm_vision/src/take_photos.py
#!/usr/bin/env python3
# -*- coding: UTF-8 -*-

import time
import datetime
import sys
import os
ros_path = '/opt/ros/kinetic/lib/python2.7/dist-packages'
if ros_path in sys.path:
   sys.path.remove(ros_path)
import cv2 as cv
sys.path.append('/opt/ros/kinetic/lib/python2.7/dist-packages')
sys.path.append('/home/domistic/Vision/Function')

from Kinect import Camera

cam = Camera()
if not cam.openCamera():
    print("camera failure")
    cam.closeCamera()
    sys.exit(1)

print("sucessful open")

today = datetime.date.today()
floder_string = "/home/domistic/Vision/Data/" + str(today)
print("目录：" + floder_string)
if not os.path.exists(floder_string):
    os.mkdir(floder_string)

pic_num = 0
cv.namedWindow("colorImg")
while(1):
    frame = cam.transColorImg()
    frame = cv.resize(frame,(int(1920/2),int(1080/2)))
    cv.imshow("colorImg",frame)
    cam.Awaken()

    if cv.waitKey(1) == ord('c'):
        today = datetime.date.today()
        now = time.strftime("%H:%M:%S")
        string = floder_string + "/" + str(today) + "-" +  str(now) + ".jpg"
        cv.imwrite(string,frame)
        pic_num += 1
        print("You have taken " + str(pic_num) + " pictures")

    if cv.waitKey(1) == ord('q'):
        cv.destroyAllWindows()
        break

cam.closeCamera()

sys.exit(0)

File: xm_vision/src/take_photos.cpp
#include <iostream>
#include <Camera.h>
#include <time.h>

std::string getTime()
{
    time_t timep;
    time (&timep);
    char tmp[64];
    strftime(tmp,sizeof(tmp), "%Y-%m-%d %H:%M:%S",localtime(&timep));
    return tmp;
}

int main()

{
    
    Vision::Camera cam;
    cv::Mat frame;
    // cv::namedWindow("windows",0);
    // cv::namedWindow("test",0);
    if(!cam.openCamera())
    {
    	std::cout<<"camera open failed"<<std::endl;
	return 0;
    }

    // frame=cv::imread("/home/domistic/pic/1.jpg");
    // cv:: resize(frame,frame,cv::Size(800,600));  
    // cv::imshow("test",frame);
    int pic_num =400;  
   // system("sudo rm     /home/domistic/Vision/data/dataset/");
   // system("sudo mkdir  /home/domistic/Vision/data/dataset");
    cv::waitKey(1000);
    while (1)
    {
	    
	    cam.transColorImg().copyTo(frame);       
        // cv::imshow("windows", frame);
        cv:: resize(frame,frame,cv::Size(0.5*frame.cols,0.5*frame.rows));
        cv::imshow("test",frame);
        char ch =cv::waitKey(1);
        if(ch == 'c')
        {
            char buf[7];
            sprintf(buf,"%06d",pic_num);
            std::string name;
            name = buf;
	    std::string time = getTime();
            std::string path = "/home/domistic/Vision/data/picture2019/"+name+"_"+time+".jpg";
            cv::imwrite(path,frame);
            pic_num++;
            std::cout<<"picture have "<<pic_num<<"@"<<time<<std::endl;
        }
        if(ch == 'q')
	    break;    
    }

    cam.closeCamera();
    return 0;

         

}
